sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/DoubleExponentialHistogramBuckets.java

```diff
@@ -48,6 +45,9 @@
   }
 
   public boolean record(double value) {
+    if (value == 0.0) {
+      throw new IllegalStateException("Illegal attempted recording of zero at bucket level.");
```

Comment:
this will potentially crash the user's app, correct? Are we really ok with that, rather than ignoring the recording, and using the ThrottlingLogger to make sure we don't spam the logs too hard?

Reply:
It's indicative of a bug if this happens. The `Handle` ensures that all zero values go towards a separate counter `zeroCount`. This avoids `Math.log(0)`. 

Up to you if we log or throw an exception, I was just aligning with what @anuraaga said here on a similar issue: https://github.com/open-telemetry/opentelemetry-java/pull/3724#discussion_r745332146

Reply:
I'm not too sure, but one difference between the two is that one is a private method but this one is public.

Perhaps my comment wasn't clear enough, by programming bug I meant a programming bug of the OTel SDK, not user app. If this is a public API that can be called by a user, and it's their problematic code that causes 0.0 to be passed, then it's a case we wouldn't throw an exception.

Reply:
This brings another issue to mind, this method should be package private, so I've got ahead and made it so. This is not a user facing method. If 0 is passed to this, it would mean there's a bug in the SDK. 

Given this info, should it be a log or an exception?

Reply:
exception is fine. I see this case is guarded by the caller. Can. you add a comment to that effect right before the line where you throw here? Thanks!

Reply:
Sounds good, I've added a comment 👍